flow/project.py

```diff
@@ -3554,6 +3836,25 @@ def _register_operations(self):
             else:
                 self._operations[name] = FlowOperation(op_func=func, **params)
 
+    def _generate_aggregates(self, operations=None):
+        """Generate aggregates for an operation provided by a user.
+        By default aggregates all the operations will be generated
+
+        :param operation:
+            Name of the operations for which aggregates are to be generated.
+        :type operation:
+            Sequence of str
+        """
+        if operations is None:
+            operations = self._groups
+
+        for op_name in operations:
+            # TODO: The logic needs to get changed in #336 from the
+            # below logic to make aggregates using group.aggregate(jobs).
+            if self._groups.get(op_name, None) is None:
+                raise ValueError(f'Invalid operation {op_name} provided by the user.')
+            self._aggregates[op_name] = [(job,) for job in self]
+
```

Comment:
This is akin to registering operations or groups. I would like to see this as a register function.